### PR TITLE
Add CircleCI config for building APK

### PR DESCRIPTION
**Pull request by hatstand**

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+      - checkout
+      - restore_cache:
+          key: jars-{{ checksum "build.gradle" }}-{{ checksum "app/build.gradle" }}
+      - run:
+          name: Download Dependencies
+          command: ./gradlew androidDependencies
+      - save_cache:
+          key: jars-{{ checksum "build.gradle" }}-{{ checksum "app/build.gradle" }}
+          paths:
+            - ~/.gradle
```
Comment (friederbluemle) [inline .circleci/config.yml]: Just checking: Was this config file copied from somewhere (with more information on the specific steps)?
Note that Gradle does not save dependencies into this local `.gradle` directory. That's different from something like NPM/Yarn (which _does_ download dependencies locally into `node_modules` locally). In order to keep this config file short and simple, I think you can just remove the `restore_cache`/"Download Dependencies"`/`save_cache` steps, it should have little to no effect on the build time.

File: .circleci/config.yml [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+      - run:
+          command: cp app/key.properties.sample app/key.properties
+      - run:
+          name: Generate signing key
+          command: keytool -genkey -v -keystore app/debug.keystore -alias AndroidDebugKey -keyalg RSA -keysize 2048 -validity 1 -storepass android -keypass android -dname "cn=TEST, ou=TEST, o=TEST, c=TE"
+      - run:
+          name: Build APK
+          command: ./gradlew clean build
```
Comment (friederbluemle) [inline .circleci/config.yml]: Is there a need for the "clean" command here (i.e. is there any leftover from previous builds)?

File: .circleci/config.yml (hunk 1 above)
Comment (hatstand) [reply .circleci/config.yml]: It's basically the example CircleCI one. AFAIK, Gradle saves jar files to `~/.gradle` so there is a minor benefit to doing this.

File: .circleci/config.yml (hunk 2 above)
Comment (hatstand) [reply .circleci/config.yml]: Doesn't hurt.

File: .circleci/config.yml (hunk 1 above)
Comment (friederbluemle) [reply .circleci/config.yml]: Very interesting. Thanks for replying. I wasn't aware of jar files in there. It's pretty odd that they cache the `.gradle` directory (which includes Gradle version specific stuff) but don't base it on the version of Gradle wrapper (which is defined in `gradle/wrapper/gradle-wrapper.properties`).
Were you able to measure the benefit somehow?

Comment (hatstand) [reply .circleci/config.yml]: I doubt the version of gradle matters as the dependent jars won't change.

I presume CircleCI think it's worth it

**Pull request merged**